lib/runtime/procs/read.js

```diff
@@ -81,6 +81,13 @@ var Read = source.extend({
             throw err;
         }
 
+        this.procName += ' ' + adapter.name;
+        var adapterAllowedOptions = this.adapter.allowedOptions();
+        if (adapterAllowedOptions) {
+            var allAllowedOptions = AdapterRead.commonOptions.concat(adapterAllowedOptions);
```

Comment:
All things being equal, I don't think it makes sense to mandate that all adapters necessarily accept the common read options. Although slightly more work for the adapter implementation, it seems safer if the subclass allowedOptions explicitly opts in to the common read options.